pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -194,15 +194,21 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 			c.Logger.Errorf("Error creating EventListener Service: %s", err)
 			return err
 		}
-	} else if !reflect.DeepEqual(oldService, service) {
-		// Update the EventListener Service
-		oldService.Spec = service.Spec
-		_, err = c.KubeClientSet.CoreV1().Services(el.Namespace).Update(oldService)
-		if err != nil {
-			c.Logger.Errorf("Error updating EventListener Service: %s", err)
-			return err
+	} else {
+		// clusterIP cannot be updated once set. So, ignore it when comparing the serviceSpecs
+		if oldService.Spec.ClusterIP != "" {
+			service.Spec.ClusterIP = oldService.Spec.ClusterIP
+		}
+		if !reflect.DeepEqual(oldService.Spec, service.Spec) {
+			// Update the EventListener Service
+			oldService.Spec = service.Spec
```

Comment:
This is setting the old spec to the new spec (same as before), which doesn't have the clusterIP value. Easy fix should be to only set the necessary values piecemeal. This was mostly a convenient recycling of the deployment reconciliation because there is more values nested within that spec. The only thing that matter for the service is likely labels.